testspec/src/14_pubkey_sig.rst

```diff
@@ -1440,6 +1440,143 @@ constraints for this test case are:
    |                        |    #. E >= 2                                                            |
    +------------------------+-------------------------------------------------------------------------+
 
+SPHINCS+
+--------
+
+The implementation is tested for correctness using the Known Answer Test vectors
+demanded by the NIST submission and provided by the reference implementation.
+Given SPHINCS+' performance characteristics, each supported algorithm
+parameterization gets just a single KAT test.
+
+Along with those integration tests Botan comes with a number of unit tests whose
+vectors were also extracted from intermediate results of the reference
+implementation. Particularly, the SPHINCS+-specific implementation of WOTS+ and
+FORS is covered by those unit tests.
+
+Additionally, Botan has implementation-specific test cases. Those ensure the
+interoperability of the algorithm when using Botan's generic API for public key
+algorithms. These test cases are equal for all public key schemes and are
+therefore not discussed in detail in this chapter.
+
+.. table::
+   :class: longtable
+   :widths: 20 80
+
+   +------------------------+-------------------------------------------------------------------------+
+   | **Test Case No.:**     | PKSIG-SPHINCS+-1                                                        |
+   +========================+=========================================================================+
+   | **Type:**              | Known Answer Tests                                                      |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Description:**       | Uses the KAT vectors of SPHINCS+' reference implementation as           |
+   |                        | specified in the NIST submission.                                       |
```

Comment:
Precise reference please

Reply:
I referenced [this document](https://csrc.nist.gov/CSRC/media/Projects/Post-Quantum-Cryptography/documents/example-files/api-notes.pdf) where NIST specified how test vectors for the submitted algorithms are to be generated.

Each reference implementation compiles a bunch of binaries that generate KATs according to this spec. Hence, referencing the actual KATs is not feasible as they are generated on demand by the consumer.

Reply:
OK, giving a reference to the NIST API specification is also good but since he reference implementation and also the NIST submission are kind of moving targets (depending on the round), I wanted to have a specific reference to what has been used, like a reference to https://sphincs.org/data/sphincs+-round3-submission-nist.zip for now (or probably better a version on NIST's website of this).

Reply:
I think the "official" round three submission ZIP is also not what we want, as the development of the algorithm continued after that. We based our implementation on: https://github.com/sphincs/sphincsplus/commit/06f42f47491085ac879a72b486ca8edb10891963

So let's maybe reference this particular commit as:

```
Test vectors were generated using the SPHINCS+ reference implementation [1] in accordance to NIST's KAT specifications [2].

[1] https://github.com/sphincs/sphincsplus/commit/06f42f47491085ac879a72b486ca8edb10891963
[2] https://csrc.nist.gov/CSRC/media/Projects/Post-Quantum-Cryptography/documents/example-files/api-notes.pdf
```